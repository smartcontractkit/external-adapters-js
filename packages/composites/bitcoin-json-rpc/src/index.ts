import { makeExecute } from './adapter'
import { expose } from '@chainlink/ea-bootstrap'

const NAME = 'BITCOIN_JSON_RPC'
export = { NAME, makeExecute, ...expose(NAME, makeExecute()) }
