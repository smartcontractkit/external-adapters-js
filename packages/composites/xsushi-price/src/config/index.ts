import { Config as BaseConfig } from '@chainlink/types'
import { Requester, util } from '@chainlink/ea-bootstrap'
import { ethers } from 'ethers'

export const ENV_XSUSHI_ADDRESS = 'XSUSHI_ADDRESS'
export const ENV_ETHEREUM_RPC_URL = 'ETHEREUM_RPC_URL'
export const ENV_FALLBACK_RPC_URL = 'RPC_URL'

export const DEFAULT_XSUSHI_ADDRESS = '0x8798249c2E607446EfB7Ad49eC89dD1865Ff4272'
export const DEFAULT_ENDPOINT = 'price'
export const NAME = 'XSUSHI_PRICE'

export type Config = BaseConfig & {
  xsushiAddress: string
  provider: ethers.providers.Provider
}

export const makeConfig = (prefix?: string): Config => {
  const rpcUrl = util.getRequiredEnvWithFallback(
    ENV_ETHEREUM_RPC_URL,
    [ENV_FALLBACK_RPC_URL],
    prefix,
  )

  return {
    ...Requester.getDefaultConfig(prefix),
    xsushiAddress: util.getEnv(ENV_XSUSHI_ADDRESS, prefix) || DEFAULT_XSUSHI_ADDRESS,
    provider: new ethers.providers.JsonRpcProvider(rpcUrl),
    defaultEndpoint: DEFAULT_ENDPOINT,
  }
}
