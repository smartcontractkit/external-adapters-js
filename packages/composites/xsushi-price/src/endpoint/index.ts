import type { TInputParameters as PriceInputParameters } from './price'
import type { TInputParameters as RatioInputParameters } from './ratio'
import type { TInputParameters as SushiInputParameters } from './sushi'

export type TInputParameters = PriceInputParameters | RatioInputParameters | SushiInputParameters

export * as price from './price'
export * as ratio from './ratio'
export * as sushi from './sushi'
