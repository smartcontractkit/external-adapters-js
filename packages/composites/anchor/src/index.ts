import { expose } from '@chainlink/ea-bootstrap'
import { endpointSelector, makeExecute } from './adapter'
import { makeConfig, NAME } from './config'

/**
 * NOTE: This adapter could probably be made more general in the future.  All it is really doing is fetching an intermediary token price
 * using the Token Allocation adapter and handling the price after that.
 */

const adapterContext = { name: NAME }

const { server } = expose(adapterContext, makeExecute(), undefined, endpointSelector)

export { NAME, makeConfig, makeExecute, server }
