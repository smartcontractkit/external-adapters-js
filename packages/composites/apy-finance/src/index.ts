import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute } from './adapter'
import { makeConfig } from './config'

const NAME = 'APY-Finance'

export = { NAME, makeConfig, makeExecute, ...expose(NAME, makeExecute()) }
