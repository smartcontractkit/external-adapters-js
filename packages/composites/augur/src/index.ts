import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute } from './adapter'
import { makeConfig } from './config'

const NAME = 'AUGUR'

export = { NAME, makeConfig, makeExecute, ...expose(NAME, makeExecute()) }
