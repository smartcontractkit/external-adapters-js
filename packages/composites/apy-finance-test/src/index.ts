import { expose, ServerInstance } from '@chainlink/external-adapter-framework'
import { Adapter } from '@chainlink/external-adapter-framework/adapter'
import { allocations, tvl } from './endpoint'
import { config } from './config'

export const adapter = new Adapter({
  defaultEndpoint: tvl.name,
  name: 'APY_FINANCE_TEST',
  config,
  endpoints: [allocations, tvl],
})

export const server = (): Promise<ServerInstance | undefined> => expose(adapter)
