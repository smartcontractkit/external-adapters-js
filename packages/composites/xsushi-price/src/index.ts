import { expose } from '@chainlink/ea-bootstrap'
import { endpointSelector, makeExecute } from './adapter'
import { makeConfig, NAME } from './config'
import * as types from './endpoint'

const adapterContext = { name: NAME }

const { server } = expose(adapterContext, makeExecute(), undefined, endpointSelector)
export { NAME, makeExecute, makeConfig, server, types }
