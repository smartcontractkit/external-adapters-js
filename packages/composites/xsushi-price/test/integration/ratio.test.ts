import { AdapterRequest } from '@chainlink/types'
import request from 'supertest'
import process from 'process'
import nock from 'nock'
import http from 'http'
import { server as startServer } from '../../src'
import { mockEthereumResponseSuccess } from './fixtures'

let oldEnv: NodeJS.ProcessEnv

beforeAll(() => {
  oldEnv = JSON.parse(JSON.stringify(process.env))
  process.env.CACHE_ENABLED = 'false'
  process.env.ETHEREUM_RPC_URL = process.env.ETHEREUM_RPC_URL || 'http://localhost:8545'
  process.env.API_VERBOSE = true
  if (process.env.RECORD) {
    nock.recorder.rec()
  }
})

afterAll(() => {
  process.env = oldEnv
  if (process.env.RECORD) {
    nock.recorder.play()
  }

  nock.restore()
  nock.cleanAll()
  nock.enableNetConnect()
})

describe('execute', () => {
  const id = '1'
  let server: http.Server
  let req: SuperTest<Test>

  beforeAll(async () => {
    server = await startServer()
    req = request(`localhost:${(server.address() as AddressInfo).port}`)
    process.env.CACHE_ENABLED = 'false'
  })

  afterAll((done) => {
    server.close(done)
  })

  describe('empty request', () => {
    const data: AdapterRequest = {
      id,
      data: {
        endpoint: 'ratio',
      },
    }

    it('should return success', async () => {
      mockEthereumResponseSuccess()

      const response = await req
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })
})
