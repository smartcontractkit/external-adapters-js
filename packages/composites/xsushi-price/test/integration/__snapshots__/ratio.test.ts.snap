// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute empty request should return success 1`] = `
{
  "data": "1196870372476465755",
  "jobRunID": "1",
  "statusCode": 200,
}
`;
